antiweb.py

```diff
@@ -2825,21 +2889,112 @@ def write(path, fname, output, token, warnings, index, index_rst, recursive, con
+   :param directory: The dedicated report directory.
+   :param file: The file part from the -output parameter.
+   :return: The absolute path of the output file.
+    """
+#@include(get_output_file_name) 
+#@(get_output_file_name doc)
+    out_file = ""
+
+    if not directory:
+        if file:
+            #check if the file has an extension. 
+            #true: the output parameter is treated as the outputfile name
+            #otherwise it is treated as the documentation directory
+            file_name, file_extension = os.path.splitext(file)
+            if file_extension:
+                out_file = os.path.join(working_dir, directory, file_name + ".rst")
+            else:
+                #the output parameter refers to a directory. the directory is combined with
+                #the input file name and optionally with the working_dir.
+                out_file = create_output_file_name(working_dir, input_file_name,  file_name)
+    else:
+        #we got a directory from the -output parameter. if the output parameter only contains a directory 
+        #the input file name is combined with the directory. otherwise the output parameter contains an entire
+        #path,  then the output file name equals the output parameter.
+        out_file = output
+        
+        if not file:
+            #we only have a directory, the directory should be combined with the input file name
+            out_file = create_output_file_name(working_dir,  input_file_name,  directory)
+    
+    return os.path.abspath(out_file)
+
+#@(get_output_file_name)
+
+#@cstart(create_output_file_name)
+
+def create_output_file_name(working_dir, input_file_name, directory):
+#@start(create_output_file_name doc)
+    """
+.. py:method:: create_output_file_name(working_dir, input_file_name, directory)
+   
+  Computes the path to the output file name. The ".rst" suffix is added to the input file name.
+  Then the paths of the working_dir and the directory are combined with the input file name.
+             
+  :param working_dir: Current working directory.
+  :param input_file_name: The name of the file which should be processed.
+  :param directory: The dedicated report directory.
+  :return: The path of the output file.
+    """
+#@include(create_output_file_name) 
+#@(create_output_file_name doc)    
+    input_rst = os.path.splitext(input_file_name)[0] + ".rst"
+    input_name = os.path.split(input_rst)[1]
+    out_file = os.path.join(working_dir, directory,  input_name)
+    return out_file
+#@(create_output_file_name)
+
+#@cstart(create_doc_directory)
+
+def create_doc_directory(out_file):
+#@start(create_doc_directory doc)
+    """
+.. py:method:: create_doc_directory(out_file)
+    
+   Creates the documentation directory if it not yet exists.
+   If an error occurs the program exits.
+           
+   :param out_file: The path to the output file.
+    """
+#@include(create_doc_directory) 
+#@(create_doc_directory doc)        
+    try:
```

Comment:
In the documentation output, the code for this function has line numbers, but the code sections for *get_output_file_name* and *create_output_file_name* do not. Why? !fix